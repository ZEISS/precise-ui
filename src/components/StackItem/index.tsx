import * as React from 'react';
import styled from '../../utils/styled';
import { StandardProps } from '../../common';

export interface StackItemProps extends StandardProps {
  /**
   * Sets the items's children.
   */
  children?: React.ReactNode;
  /**
   * Sets the width of the item.
   */
  width?: string;
  /**
   * Sets the height of the item.
   */
  height?: string;
  /**
   * Determines if the item should not fill the remaining space.
   */
  nofill?: boolean;
}

const StackChild = styled.div`
  flex-grow: ${(props: StackItemProps) => (props.nofill ? '0' : '1')};
  min-width: ${(props: StackItemProps) => props.width || 'auto'};
  height: ${(props: StackItemProps) => props.height || 'auto'};
`;

/**
 * The stack item displays a single item within a stack panel.
 */
export class StackItem extends React.PureComponent<StackItemProps> {
  render() {
    const { children, ...props } = this.props;
    return <StackChild {...props}>{children}</StackChild>;
  }
}
