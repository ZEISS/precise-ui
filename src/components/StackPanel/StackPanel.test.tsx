import * as React from 'react';
import * as enzyme from 'enzyme';
import { StackPanel, StackPanelDirection } from './';
import { StackItem } from '../StackItem';

describe('<StackPanel />', () => {
  it('should render the default StackPanel', () => {
    const wrapper = enzyme.mount(<StackPanel />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the StackPanel with three items', () => {
    const wrapper = enzyme.mount(
      <StackPanel>
        <StackItem>First</StackItem>
        <StackItem>Second</StackItem>
        <StackItem>Third</StackItem>
      </StackPanel>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the StackPanel with explicit widths and wrapping', () => {
    const wrapper = enzyme.mount(
      <StackPanel wrap>
        <StackItem width="440px">First</StackItem>
        <StackItem width="300px">Second</StackItem>
        <StackItem width="40%">Third</StackItem>
      </StackPanel>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the StackPanel with explicit withs and no wrapping', () => {
    const wrapper = enzyme.mount(
      <StackPanel>
        <StackItem width="100px">First</StackItem>
        <StackItem width="100px">Second</StackItem>
      </StackPanel>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the StackPanel bottom to top', () => {
    const wrapper = enzyme.mount(
      <StackPanel direction={StackPanelDirection.bottomToTop}>
        <StackItem>First</StackItem>
        <StackItem>Second</StackItem>
      </StackPanel>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the StackPanel right to left', () => {
    const wrapper = enzyme.mount(
      <StackPanel direction={StackPanelDirection.rightToLeft}>
        <StackItem>First</StackItem>
        <StackItem>Second</StackItem>
      </StackPanel>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the StackPanel top to bottom with explicit heights', () => {
    const wrapper = enzyme.mount(
      <StackPanel direction={StackPanelDirection.topToBottom} style={{ height: '400px' }}>
        <StackItem height="20%">First</StackItem>
        <StackItem height="60%">Second</StackItem>
        <StackItem height="20%">Third</StackItem>
      </StackPanel>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
