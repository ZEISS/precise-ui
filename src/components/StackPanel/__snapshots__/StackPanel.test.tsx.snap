// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StackPanel /> should render the StackPanel bottom to top 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: auto;
  height: auto;
}

<StackPanel
  direction="bottom"
>
  <styled.div
    dir="column-reverse"
    wrapping="nowrap"
  >
    <StyledComponent
      dir="column-reverse"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: ",
              [Function],
              ";
  flex-wrap: ",
              [Function],
              ";
  justify-content: flex-start;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      wrapping="nowrap"
    >
      <div
        className="c0"
        dir="column-reverse"
      >
        <StackItem>
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  flex-grow: ",
                      [Function],
                      ";
  min-width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="c1"
              >
                First
              </div>
            </StyledComponent>
          </styled.div>
        </StackItem>
        <StackItem>
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  flex-grow: ",
                      [Function],
                      ";
  min-width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="c1"
              >
                Second
              </div>
            </StyledComponent>
          </styled.div>
        </StackItem>
      </div>
    </StyledComponent>
  </styled.div>
</StackPanel>
`;

exports[`<StackPanel /> should render the StackPanel right to left 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: auto;
  height: auto;
}

<StackPanel
  direction="right"
>
  <styled.div
    dir="row-reverse"
    wrapping="nowrap"
  >
    <StyledComponent
      dir="row-reverse"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: ",
              [Function],
              ";
  flex-wrap: ",
              [Function],
              ";
  justify-content: flex-start;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      wrapping="nowrap"
    >
      <div
        className="c0"
        dir="row-reverse"
      >
        <StackItem>
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  flex-grow: ",
                      [Function],
                      ";
  min-width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="c1"
              >
                First
              </div>
            </StyledComponent>
          </styled.div>
        </StackItem>
        <StackItem>
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  flex-grow: ",
                      [Function],
                      ";
  min-width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="c1"
              >
                Second
              </div>
            </StyledComponent>
          </styled.div>
        </StackItem>
      </div>
    </StyledComponent>
  </styled.div>
</StackPanel>
`;

exports[`<StackPanel /> should render the StackPanel top to bottom with explicit heights 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: auto;
  height: 20%;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: auto;
  height: 60%;
}

<StackPanel
  direction="top"
  style={
    Object {
      "height": "400px",
    }
  }
>
  <styled.div
    dir="column"
    style={
      Object {
        "height": "400px",
      }
    }
    wrapping="nowrap"
  >
    <StyledComponent
      dir="column"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: ",
              [Function],
              ";
  flex-wrap: ",
              [Function],
              ";
  justify-content: flex-start;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      style={
        Object {
          "height": "400px",
        }
      }
      wrapping="nowrap"
    >
      <div
        className="c0"
        dir="column"
        style={
          Object {
            "height": "400px",
          }
        }
      >
        <StackItem
          height="20%"
        >
          <styled.div
            height="20%"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  flex-grow: ",
                      [Function],
                      ";
  min-width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              height="20%"
            >
              <div
                className="c1"
                height="20%"
              >
                First
              </div>
            </StyledComponent>
          </styled.div>
        </StackItem>
        <StackItem
          height="60%"
        >
          <styled.div
            height="60%"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  flex-grow: ",
                      [Function],
                      ";
  min-width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              height="60%"
            >
              <div
                className="c2"
                height="60%"
              >
                Second
              </div>
            </StyledComponent>
          </styled.div>
        </StackItem>
        <StackItem
          height="20%"
        >
          <styled.div
            height="20%"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  flex-grow: ",
                      [Function],
                      ";
  min-width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              height="20%"
            >
              <div
                className="c1"
                height="20%"
              >
                Third
              </div>
            </StyledComponent>
          </styled.div>
        </StackItem>
      </div>
    </StyledComponent>
  </styled.div>
</StackPanel>
`;

exports[`<StackPanel /> should render the StackPanel with explicit widths and wrapping 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 440px;
  height: auto;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 300px;
  height: auto;
}

.c3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 40%;
  height: auto;
}

<StackPanel
  wrap={true}
>
  <styled.div
    dir="row"
    wrapping="wrap"
  >
    <StyledComponent
      dir="row"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: ",
              [Function],
              ";
  flex-wrap: ",
              [Function],
              ";
  justify-content: flex-start;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      wrapping="wrap"
    >
      <div
        className="c0"
        dir="row"
      >
        <StackItem
          width="440px"
        >
          <styled.div
            width="440px"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c3",
                    "rules": Array [
                      "
  flex-grow: ",
                      [Function],
                      ";
  min-width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              width="440px"
            >
              <div
                className="c1"
                width="440px"
              >
                First
              </div>
            </StyledComponent>
          </styled.div>
        </StackItem>
        <StackItem
          width="300px"
        >
          <styled.div
            width="300px"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c3",
                    "rules": Array [
                      "
  flex-grow: ",
                      [Function],
                      ";
  min-width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              width="300px"
            >
              <div
                className="c2"
                width="300px"
              >
                Second
              </div>
            </StyledComponent>
          </styled.div>
        </StackItem>
        <StackItem
          width="40%"
        >
          <styled.div
            width="40%"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c3",
                    "rules": Array [
                      "
  flex-grow: ",
                      [Function],
                      ";
  min-width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              width="40%"
            >
              <div
                className="c3"
                width="40%"
              >
                Third
              </div>
            </StyledComponent>
          </styled.div>
        </StackItem>
      </div>
    </StyledComponent>
  </styled.div>
</StackPanel>
`;

exports[`<StackPanel /> should render the StackPanel with explicit withs and no wrapping 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 100px;
  height: auto;
}

<StackPanel>
  <styled.div
    dir="row"
    wrapping="nowrap"
  >
    <StyledComponent
      dir="row"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: ",
              [Function],
              ";
  flex-wrap: ",
              [Function],
              ";
  justify-content: flex-start;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      wrapping="nowrap"
    >
      <div
        className="c0"
        dir="row"
      >
        <StackItem
          width="100px"
        >
          <styled.div
            width="100px"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  flex-grow: ",
                      [Function],
                      ";
  min-width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              width="100px"
            >
              <div
                className="c1"
                width="100px"
              >
                First
              </div>
            </StyledComponent>
          </styled.div>
        </StackItem>
        <StackItem
          width="100px"
        >
          <styled.div
            width="100px"
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  flex-grow: ",
                      [Function],
                      ";
  min-width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              width="100px"
            >
              <div
                className="c1"
                width="100px"
              >
                Second
              </div>
            </StyledComponent>
          </styled.div>
        </StackItem>
      </div>
    </StyledComponent>
  </styled.div>
</StackPanel>
`;

exports[`<StackPanel /> should render the StackPanel with three items 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: auto;
  height: auto;
}

<StackPanel>
  <styled.div
    dir="row"
    wrapping="nowrap"
  >
    <StyledComponent
      dir="row"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: ",
              [Function],
              ";
  flex-wrap: ",
              [Function],
              ";
  justify-content: flex-start;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      wrapping="nowrap"
    >
      <div
        className="c0"
        dir="row"
      >
        <StackItem>
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  flex-grow: ",
                      [Function],
                      ";
  min-width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="c1"
              >
                First
              </div>
            </StyledComponent>
          </styled.div>
        </StackItem>
        <StackItem>
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  flex-grow: ",
                      [Function],
                      ";
  min-width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="c1"
              >
                Second
              </div>
            </StyledComponent>
          </styled.div>
        </StackItem>
        <StackItem>
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": false,
                    "lastClassName": "c1",
                    "rules": Array [
                      "
  flex-grow: ",
                      [Function],
                      ";
  min-width: ",
                      [Function],
                      ";
  height: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="c1"
              >
                Third
              </div>
            </StyledComponent>
          </styled.div>
        </StackItem>
      </div>
    </StyledComponent>
  </styled.div>
</StackPanel>
`;

exports[`<StackPanel /> should render the default StackPanel 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<StackPanel>
  <styled.div
    dir="row"
    wrapping="nowrap"
  >
    <StyledComponent
      dir="row"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  display: flex;
  flex-direction: ",
              [Function],
              ";
  flex-wrap: ",
              [Function],
              ";
  justify-content: flex-start;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      wrapping="nowrap"
    >
      <div
        className="c0"
        dir="row"
      />
    </StyledComponent>
  </styled.div>
</StackPanel>
`;
